### ISSUE TITLE
Ability to disable/hide column sorting

### ISSUE THREAD

**Issue by stramel** [vaadin/vaadin-grid#852]
It would be great if we could add a way to disable and/or hide the column sorter.

**Comment by jouni** [comment]
I’m not sure I understand the need correctly. I mean, in both v1 and v2 sorting is opt-in, so just don’t use the `sortable` attribute (v1) or `vaadin-grid-sorter` (v2).

**Comment by stramel** [comment]
@jouni it is opt-in however, I'm trying to encompass a way for someone be able to remove sorting using a property on the encompassing component or use the property during dynamic generation of the grid to cover multiple cases

**Comment by jouni** [comment]
Alright, now I got a clearer picture.

I think we could add a `disabled` property for the `<vaadin-grid-sorter>` element.

**Comment by Saulis** [comment]
You can already add `hidden` attribute to `<vaadin-grid-sorter>` (or use for example `dom-if`), or apply `display: none` based on some custom class you apply on it. Just make sure you have some alternative header content which gets displayed while the sorter is hidden.

**Comment by stramel** [comment]
@Saulis won't adding hidden to the `vaadin-grid-sorter` hide the column heading as well?

> Also, you might want to clear the direction from the sorter when hiding it.

Good point.

**Comment by Saulis** [comment]
@stramel yes, if you can the heading nested as a child of the sorter element. You can either put the heading completely outside the sorter (which will make the sorting to be triggered only from the sorting indicator icon) or you can apply a secondary heading which gets displayed only when the `<vaadin-grid-sorter>` is hidden.

**Comment by stramel** [comment]
Alright, I guess that will cover the use case, just feels hacky, kinda like the filtering from outside the grid.

**Comment by Saulis** [comment]
We could still keep this issue open, since having i.e. a `disabled` property on the sorter could be a good addition. What do you think?

**Comment by stramel** [comment]
@Saulis Up to you guys, I think having one would be nice.
